pkg/kudoctl/packages/reader/parser_test.go

```diff
@@ -0,0 +1,70 @@
+package reader
+
+import (
+	"errors"
+	"testing"
+
+	"gotest.tools/assert"
+)
+
+const (
+	validOperator = `
+apiVersion: kudo.dev/v1beta1
+name: "first-operator"
+operatorVersion: "0.1.0"
+`
+	validParams = `
+apiVersion: kudo.dev/v1beta1
+parameters:
+  - name: param
+    default: "value"
+`
+)
+
+func TestParsePackageFile(t *testing.T) {
+	tests := []struct {
+		filePath    string
+		fileContent string
+
+		isOperator bool
+		isParam    bool
+		isTemplate bool
+
+		expectedError error
+	}{
+		{"operator.yaml", validOperator, true, false, false, nil},
+		{"params.yaml", validParams, false, true, false, nil},
```

Comment:
a test case ```{"/foo/bar/operator.yaml", validOperator, true, false, false, nil},``` is failing on this branch.

Reply:
See latest commit: I think the parser should only handle package-relative paths, not absolute file system paths. I've added code that normalises the paths passed into the parser.

Reply:
Also added a comment to parsePackageFile